lib/resources/nextflow_aws.template.config

```diff
@@ -23,13 +23,13 @@ params {
 
   genomes {
     GRCh38_hmf {
-      fasta         = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'
-      fai           = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/samtools_index/1.16/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.fai'
-      dict          = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/samtools_index/1.16/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.dict'
-      img           = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/bwa_index_image/0.7.17-r1188/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.img'
-      bwamem2_index = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/gridss_index/2.13.2/'
-      gridss_index  = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/gridss_index/2.13.2/'
-      star_index    = 's3://__S3_BUCKET_NAME__/__S3_BUCKET_REFDATA_PREFIX__/genomes/GRCh38_hmf/bwa-mem2_index/2.2.1/'
+      fasta         = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna'
+      fai           = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/samtools_index/1.16/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.fai'
+      dict          = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/samtools_index/1.16/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.dict'
+      img           = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/bwa_index_image/0.7.17-r1188/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.img'
+      bwamem2_index = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/gridss_index/2.13.2/'
+      gridss_index  = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/gridss_index/2.13.2/'
+      star_index    = 's3://{{S3_BUCKET_NAME}}/{{S3_BUCKET_REFDATA_PREFIX}}/genomes/GRCh38_hmf/bwa-mem2_index/2.2.1/'
```

Comment:
I think templating inside the CDK code is okay. 

As another option, could possibly also use the env function: https://www.nextflow.io/docs/latest/config.html#functions if the version is new enough?

Reply:
Yes - possibly thought of that.. problem is that each and every variable for the substitution has to be passed from CDK -> env -> bash -> nextflow env. And that's a lot of potential spots for misnamed variables etc and name drift. And it could evenutally be about 20 or 30 variables. Doing it in the CDK takes away some of that boilerplate was my thinking